### ISSUE TITLE
new feature: subject-wise transform

### ISSUE THREAD

**Issue by raamana** [raamana/pyradigm#4]
it might be helpful to have
- [ ] subject-wise transform (`MLDataset.apply_xfm`?) that applies a given callable (such as an ROI mask, subset selector or a non-linear transform) for each subject and returns a new set of features (of different size)
- [ ] `apply_xfm` takes in a callable and its description and returns a new dataset with new set of features

```python
from pyradigm import MLDataset

thickness = MLDataset(in_path='ADNI_thickness.csv')
pcg_thickness = thickness.apply_xfm(func=get_pcg, description = 'applying ROI mask for PCG')
pcg_median = pcg_thickness.apply_xfm(func=np.median, description='median per subject')

```

**Comment by raamana** [comment]
Application of more complex functions (requiring more than one argument) can be achieved via partial functions:

```python
from pyradigm import MLDataset
from functools import partial
import hiwenet

thickness = MLDataset(in_path='ADNI_thickness.csv')
roi_membership = read_roi_membership()
hw = partial(hiwenet, groups = roi_membership)

thickness_hiwenet = thickness.transform(func=hw, description = 'histogram weighted networks')
median_thk_hiwenet = thickness_hiwenet.transform(func=np.median, description='median per subject')

```

**Comment by raamana** [comment]
060d1ecd153a5cc23971c3772e116d0d8edc1502 fixes #4
